ClangSharp.PInvokeGenerator.Test/StructDeclarationTest.cs

```diff
@@ -218,6 +290,89 @@ public partial struct MyStruct
             await ValidateGeneratedBindings(inputContents, expectedOutputContents);
         }
 
+        [Theory]
+        [InlineData("unsigned char", "byte")]
+        [InlineData("double", "double")]
+        [InlineData("short", "short")]
+        [InlineData("int", "int")]
+        [InlineData("long long", "long")]
+        [InlineData("char", "sbyte")]
+        [InlineData("float", "float")]
+        [InlineData("unsigned short", "ushort")]
+        [InlineData("unsigned int", "uint")]
+        [InlineData("unsigned long long", "ulong")]
+        public async Task UnsafeFixedSizedBufferNonPrimitiveTest(string nativeType, string expectedManagedType)
+        {
+            var inputContents = $@"struct MyStruct
+{{
+    {nativeType} value;
+}};
+
+struct MyOtherStruct
+{{
+    MyStruct c[3];
+}};
+";
+
+            var expectedOutputContents = $@"using System.Runtime.InteropServices;
+
+namespace ClangSharp.Test
+{{
+    public unsafe partial struct MyStruct
+    {{
+        public {expectedManagedType} value;
+    }}
+
+    public unsafe partial struct MyOtherStruct
+    {{
+        public _c_e__FixedBuffer c;
+
+        public partial struct _c_e__FixedBuffer
+        {{
+            private MyStruct e0;
+            private MyStruct e1;
+            private MyStruct e2;
+
+            public ref MyStruct this[int index] => ref MemoryMarshal.CreateSpan(ref e0, 3)[index];
```

Comment:
This is the fallback generation (still for unsafe code) for when C# doesn't support the type for "proper" fixed-sized buffers.

It takes advantage of `Span` and `ref returns` to avoid pinning and allow "normal" use of the API.